### DISCUSSION TITLE
Forms Designer seems to break FormBrowse

### DISCUSSION THREAD

**Discussion by Timwi** [gitextensions/gitextensions#9654] [Q&A] [answered]
Hello! I’m trying to add mnemonics to the menu items in the main window (FormBrowse).

Unfortunately, when I do that in the VS 2022 form designer, it seems to make changes to the generated code (FormBrowse.Designer.cs) that cause it to no longer compile. Specifically, it seems to remove the assignments for some custom controls, including `RevisionGrid`, `TreeTabPage`, `fileTree`, etc.

Is there something I can do to fix this so that I can use the form designer normally?

**Comment by gerhardol** [comment]
Other than not not using the designer or just cherry picking changes, I am not aware of anything

**Comment by RussKie** [comment, marked as answer]
Without a major redesign I dont think it is feasible. It is non trivial to untangle multiple interdependencies and straighten the logic (e.g. have a look at the amount of changes involved in uplifting filters). With the amount of time I can dedicate to the project right now it is glacial pace.

Whenever I make any changes in FormBrowse I only keep the changes I need and reset the rest.

**Comment by Timwi** [comment]
Hm wow, that puts a big damper on development, but thanks for the feedback.
